R/a11y_panel.R

```diff
@@ -0,0 +1,250 @@
+#' Accessibility panel
+#'
+#' @param dashboard_title Title of the host dashboard
+#' @param dashboard_url URL for the host dashboard
+#' @param repo_url URL for the dashboard repository
+#' @param non_accessible_components String vector containing a list of non accessible components
+#' @param specific_issues String vector containing descriptions of specific accessibility issues
+#' that have been identified as part of testing
+#' @param date_tested Date the application was last tested
+#' @param date_prepared Date the statement was prepared in it's current form
+#' @param date_reviewed Date the statement was last reviewed
+#' @param date_template_reviewed Date the underlying template was reviewed
+#' (default: 12th March 2024)
+#'
+#' @return shiny$tags$div element containing the html tags and content for the standard
+#' accessibility statement
+#' @export
+#'
+#' @examples
+#' a11y_panel(
+#'   "DfE Shiny template",
+#'   "https://department-for-education.shinyapps.io/dfe-shiny-template",
+#'   "25th April 2024",
+#'   "26th April 2024",
+#'   "2nd November 2024",
+#'   repo_url = "https://github.com/dfe-analytical-services/shiny-template",
+#' )
+a11y_panel <- function(
+    dashboard_title,
+    dashboard_url,
+    date_tested,
+    date_prepared,
+    date_reviewed,
+    date_template_reviewed = "12 March 2024",
+    repo_url = NA,
+    non_accessible_components = c(
+      "Keyboard navigation through the interactive charts is currently limited",
+      "Alternative text in interactive charts is limited to titles"
+    ),
+    specific_issues = c(
+      "Charts have non-accessible components that are inaccessible for keyboard users.",
+      "Chart tooltips are not compativle with screen reader use.",
+      "Some decorative images are not labelled appropriately as yet.",
+      "Some links are not appropriately labelled."
+    )) {
+  # Validate inputs
+  date_tested <- validate_date(date_tested)
+  date_prepared <- validate_date(date_prepared)
+  date_reviewed <- validate_date(date_reviewed)
+  date_template_reviewed <- validate_date(date_template_reviewed)
+  validate_dashboard_url(dashboard_url)
+  if (!is_valid_repo_url(repo_url)) {
+    stop(repo_url, " is not a valid repository url")
+  }
+  shiny::tags$div(
+    "Accessibility",
+    shiny::tags$div(
+      shiny::tags$h1(paste0("Accessibility statement for ", dashboard_title)),
+      shiny::tags$p(
+        "This accessibility statement applies to the",
+        dashboard_url,
```

Comment:
Having the URL embedded here reads a bit oddly and might be irritating for someone using a screen reader:

![image](https://github.com/user-attachments/assets/9e015f92-fb91-43b0-b278-4af71f2256d8)

Reply:
That's the standard pattern I've seen for accessibility statements, I think it's so the URL in scope is completely explicit.

As it's not marked up as a link, a user will only get it when going through the text so I think this is right as is

Reply:
EES as an example
![image](https://github.com/user-attachments/assets/b73d1dac-1824-4743-8e68-bdd51b5a16b3)